src/imperative/imperative_utils.cc

```diff
@@ -120,6 +121,18 @@ void InvokeOperator(const nnvm::IndexedGraph& idx,
       *arrays[eid] = NDArray();
     }
   }
+  if (callback) {
+    for (size_t i = 0; i < ndinputs.size(); ++i) {
+      NDArray* cpy = new NDArray(*ndinputs[i]);
+      callback(node.source->attrs.name.c_str(), node.source->op()->name.c_str(),
```

Comment:
above you mention that cb takes one string and a array handle. 
here i see 2 strings, am i missing anything here?

Reply:
this is still wip, the comment is wrong and needs to be cleaned up , it takes 2 strings.